src/groups/mqb/mqba/mqba_clientsession.cpp

```diff
@@ -339,6 +339,479 @@ struct BuildAckOverflowFunctor {
 
 }  // close unnamed namespace
 
+class ClientSessionDestructGuard {
```

Comment:
I don't think this class is a guard, it's not really protecting anything. I think we can document that it's a collection of helper functions for avoiding the destruction bug though.

```suggestion
struct ClientSessionUtil {
```